src/parse.c

```diff
@@ -5245,13 +5245,51 @@ static class_t p_entity_class(void)
    }
 }
 
+static ident_t p_entity_designator()
+{
+   // entity_tag [ signature ]
+   ident_t id = p_identifier();
+
+   if (peek() == tLSQUARE) {
+      type_t constraints = p_signature();
+   }
+   return id;
+}
+
+static ident_list_t *p_entity_name_list(void)
+{
+   // entity_designator { , entity_designator } | others | all
+   ident_list_t *result = NULL;
+
+   switch (peek()) {
+   // LRM-2019 7.2 - Attribute Specification
+   case tOTHERS:
+      // Attribute applies to the names entities of the class declared in the
+      // immediately enclosing declarative part, provided that each such entity
+      // is not explicitly named in the entity name list of the previous
+      // attribute specification for the given attribute. Error if this isn't
+      // the last attribute for the given entity class.
+   case tALL:
+      // Attribute applies to all named entities of the class declared in the
+      // immediately enclosing declarative part. Error if this isn't the last
+      // attribute for the given entity class.
+      break;
```

Comment:
Probably needs a call to `consume(tALL)` here and similarly for `tOTHERS` above, otherwise you'll immediately get an unexpected token error for the next token?

Reply:
Agreed.  I'll do that, but what I was hoping to figure out here is how to convey the information that the target is `others` or `all` beyond the scope of this parsing function.

Reply:
For now it can probably just use the hack that `p_instantiation_list` does and return NULL for `others` and the magic identifier `well_known(W_ALL)`. Longer term I want to rewrite most of the attribute handling code as there are multiple issues with it beyond just this one. I suspect the right thing to do is add a `T_DESIGNATOR` AST node to hold it but I'm not sure exactly how it should work.